### DISCUSSION TITLE
关于本项目的用途的一点疑问

### DISCUSSION THREAD

**Discussion by cylind** [XIU2/CloudflareSpeedTest#356] [讨论]
开发者，你好
我看issue的时候发现，你说你开发这个工具的目的是为了寻找较快的cf cdn，然后改写host，以便更快地访问常去的国外网站。
改写host的意思是通过改写本机hosts文件，将cf cdn的ip与目标网站的域名相映射吗？比如在hosts文件中添加 `1.1.1.1 www.coursera.org`
另外，不管目标网站是否托管于cloudflare都可以通过这样的方式访问吧？
这样改写host的方法也是由cf cdn节点来中转访问流量的吧？

**Comment by XIU2** [comment]
只能用于使用 **Cloudflare CDN 的网站域名**。  

因为 Cloudflare CDN 用了**任播技术**，所以使用 Cloudflare CDN 的网站域名都可以指向任意 Cloudflare CDN 的 IP。
访客也可以自己通过任意 Cloudflare CDN 的 IP 访问该网站（必须用的是 Cloudflare CDN）。

****

当然你说的也不是不行，比如先用 Cloudflare Workers 搭建一个**反向代理**服务（免费的，不需要自己服务器），然后指向速度快的 CDN IP，也就实现你说的：
- 不管目标网站是否托管于cloudflare都可以通过这样的方式访问
- 由cf cdn节点来中转访问流量

**Comment by cylind** [comment]
感谢回复。
还真是，我刚才尝试了常用的几个网站，ddrk.me，www.icourse163.org，github.com，偶尔都会出现访问不了（能访问的话估计是浏览器缓存或DNS缓存），并提示ssl协议不受支持，估计是网站启用https。
另外，你说的利用cloudflare worker搭建反向代理，听起来很棒，我去试试。

**Comment by XIU2** [comment]
只有使用 Cloudflare CDN 的网站才能指向 Cloudflare CDN 的 IP 。

你说的这三个网站，**都没有使用 Cloudflare CDN**，你强行指向 Cloudflare CDN IP 去访问，肯定会报错。
和网站是否使用 HTTPS 无关，当 Cloudflare CDN 没有找到这个域名（自己的数据库），就会报这个错误。

**Comment by cylind** [comment]
话说要怎么知道一个网站用没用cloudflare cdn呢？

**Comment by XIU2** [comment]
### 方法一：

在网站域名后面加上 `/cdn-cgi/trace` 并访问就会看到一串信息。
> 如果提示 404 之类的错误，代表没有使用 Cloudflare CDN。

例如 `https://cloudflare.com` 就是：https://cloudflare.com/cdn-cgi/trace

****

### 方法二：

查一下这个网站的 IP 归属。

比如去这里输入域名查看：https://www.ipip.net/ip.html

**Comment by cylind** [comment]
> 当然你说的也不是不行，比如先用 Cloudflare Workers 搭建一个反向代理服务

这个不太好弄，除了维基百科还代理得不错，其他的要不是页面乱码，就是访问量过大（google、youtube等），直接被判为bot。还是正向代理好，比如cloudflare warp。

**Comment by XIU2** [comment]
嗯，还有其他问题吗？
如果没有的话，就关闭该 Issues 吧。
